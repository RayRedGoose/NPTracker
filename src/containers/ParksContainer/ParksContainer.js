import './ParksContainer.scss'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getData } from 'apiCalls'
import { addParks } from 'redux/actions'
import ParkCardPreview from 'containers/ParkCardPreview/ParkCardPreview'

export class ParksContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLoaded: false,
      error: null
    }
  }

  componentDidMount() {
    if (!this.props.parks.length) this.fetchParks()
  }

  fetchParks = async () => {
    try {
      const parks = await getData('parks', 'park')
      this.setState({isLoaded: true})
      this.props.addParks(parks)
    }
    catch (error) {
      this.setState({
        isLoaded: false,
        error: error.message
      })
    }
  }

  createCards = () => {
    const { parks, type, plannedParks } = this.props
    const planned = parks.filter(park => plannedParks.includes(park.name))
    const data = (type === 'all') ? parks : planned
    return data.map(park => (
      <ParkCardPreview
        key={park.id}
        id={park.id}
        image={park.images[0].url}
        fullName={park.fullName}
        name= {park.name}
        states={park.states}/>
    ))
  }

  render() {
    return (
      <section className="parks-content">
        { this.createCards()}
      </section>
    )
  }
}

export const mapStateToProps = ({parks, plannedParks}) => ({
  parks, plannedParks
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addParks
  }, dispatch)
)

ParksContainer.propTypes = {
  type: PropTypes.string.isRequired,
  parks: PropTypes.arrayOf(PropTypes.object).isRequired,
  plannedParks: PropTypes.arrayOf(PropTypes.string).isRequired,
  addParks: PropTypes.func.isRequired
}


export default connect(mapStateToProps, mapDispatchToProps)(ParksContainer)
