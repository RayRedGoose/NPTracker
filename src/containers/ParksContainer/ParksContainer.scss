.parks-content {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 40px;
  width: 100%;
  overflow: scroll;
}
