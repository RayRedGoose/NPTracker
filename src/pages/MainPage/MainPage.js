import './MainPage.scss'
import React from 'react'
import PropTypes from 'prop-types'
import NavPanel from 'containers/NavPanel/NavPanel'
import SearchForm from 'containers/SearchForm/SearchForm'
import UserProfile from 'containers/UserProfile/UserProfile'

const MainPage = ({ Part, type }) => {
  return (
    <main className="main-page">
      <header>
        <NavPanel />
        <SearchForm />
      </header>
      <UserProfile />
      <Part type={type} />
    </main>
  )
}

MainPage.propTypes = {
  Part: PropTypes.elementType.isRequired,
  type: PropTypes.string
}

export default MainPage
