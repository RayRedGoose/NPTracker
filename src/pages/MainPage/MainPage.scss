.main-page {
  display: grid;
  grid-template-areas: 'tabs user' 'main user';
  grid-template-columns: 4fr 1fr;
  grid-template-rows: 1fr 9fr;
  &>nav {
    grid-area: tabs;
  }
  &>aside {
    grid-area: user;
  }
  &>section {
    grid-area: main;
  }
}
