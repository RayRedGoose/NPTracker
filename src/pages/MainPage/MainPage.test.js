import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';
import ParksContainer from 'containers/ParksContainer/ParksContainer'

it("should match snapshot with all data passed in correctly", () => {
  const page = shallow(
    <MainPage Part={ParksContainer} type="all" />
  )
  expect(page).toMatchSnapshot();
});
