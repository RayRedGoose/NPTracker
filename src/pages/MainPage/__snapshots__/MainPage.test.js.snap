// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with all data passed in correctly 1`] = `
<main
  className="main-page"
>
  <Connect(NavPanel) />
  <Connect(UserProfile) />
  <Connect(ParksContainer)
    type="all"
  />
</main>
`;
